docs/src/appendix/migrating-from-chiseltest.md

```diff
@@ -0,0 +1,103 @@
+
+## Getting Started
+
+The developers of ChiselTest have maintained some amount of compatibility between ChiselTest and newer versions of Chisel.
+This relies on a forked version of the SFC.
+Use of ChiselTest with Chisel 6 or later will prevent the usage of new Chisel 6 features.
+It is not expected that new versions of Chisel will be compatible with the SFC.
+
+We recommend using the latest minor version of Chisel 5 and ChiselTest 5 for migrating. At the time of writing this is Chisel v5.1.0 and ChiselTest v5.0.2, but please check for later versions.
+
+## Migration
+
+ChiselSim provides a minimal `peek`, `poke`, `expect`, and `step` API, similar to that of ChiselTest.
+You use ChiselSim by importing it `import chisel3.simulator.EphemeralSimulator._`, and using its `simulate` method which is similar to ChiselTest's `test`.
+At present, it does not have any integration with ScalaTest, so users should use any ScalaTest APIs directly.
+
+For example, given a simple design (typically in `src/main/scala`):
+
+```scala mdoc:silent
+import chisel3._
+class MyModule extends Module {
+  val io = IO(new Bundle {
+    val in = Input(UInt(16.W))
+    val out = Output(UInt(16.W))
+  })
+
+  io.out := RegNext(io.in)
+}
+```
+
+The legacy ChiselTest way to test this would be with a `ChiselScalatestTester` in `src/test/scala`:
+
+<!-- This cannot be mdoc because we do not compile against chiseltest -->
+```scala
+import chisel3._
+import chiseltest._
+import org.scalatest.flatspec.AnyFlatSpec
+
+class MyModuleSpec extends AnyFlatSpec with ChiselScalatestTester {
+  behavior of "MyModule"
+  it should "do something" in {
+    test(new MyModule) { c =>
+      c.io.in.poke(0.U)
+      c.clock.step()
+      c.io.out.expect(0.U)
+      c.io.in.poke(42.U)
+      c.clock.step()
+      c.io.out.expect(42.U)
+      println("Last output value : " + c.io.out.peek().litValue)
+    }
+  }
+}
+```
+
+This can be rewritten using ChiselSim as follows:
+```scala mdoc:silent
+import chisel3._
+import chisel3.simulator.EphemeralSimulator._
+import org.scalatest.flatspec.AnyFlatSpec
+
+class MyModuleSpec extends AnyFlatSpec {
+  behavior of "MyModule"
+  it should "do something" in {
+    simulate(new MyModule) { c =>
+      c.io.in.poke(0.U)
+      c.clock.step()
+      c.io.out.expect(0.U)
+      c.io.in.poke(42.U)
+      c.clock.step()
+      c.io.out.expect(42.U)
+      println("Last output value : " + c.io.out.peek().litValue)
+    }
+  }
+}
+```
+
+For both ChiselTest and ChiselSim, you will typically run this with `sbt test` or some way of running tests.
+The output from ChiselSim will look something like the following:
+
+```scala mdoc
+// This is how one can run a ScalaTest Spec manually, typically one would use "sbt test"
+org.scalatest.nocolor.run(new MyModuleSpec)
+```
+
+ChiselSim also does not currently have any support for `fork`-`join`, so any tests using those constructs will need to be rewritten in a single-threaded manner.
```

Comment:
It feels like this could be the start of a new section, called something like "Missing Features", which details all of the ChiselTest features that aren't yet supported. We should probably give a plan for each one: is it going to be supported in the future? Should it be replaced with a different style of code?

Reply:
Yeah that could be a useful section. For things with different style, I think they get larger sections since you'll want the code example for "before" and "after". But yeah having a plan and giving pointers would be useful since perhaps people will want to contribute to some of these. For example, formal verification needs more work on CIRCT but maybe with a pointer some people could contribute.

Reply:
In any case, I think I'll get this merged to have something to point to and then we should continue iterating on this doc.